nvflare/app_common/statistics/numeric_stats.py

```diff
@@ -233,27 +237,50 @@ def filter_numeric_features(ds_features: Dict[str, List[Feature]]) -> Dict[str,
     return numeric_ds_features
 
 
-def aggregate_centroids(metrics: Dict[str, Dict[str, Dict]], g_digest: dict) -> dict:
+def merge_quantiles(metrics: Dict[str, Dict[str, Dict]], g_digest: dict) -> dict:
+    TDigest = check_and_import_tdigest()
+    if TDigest is None:
+        return g_digest
+
+    from fastdigest import TDigest
+
     for ds_name in metrics:
         if ds_name not in g_digest:
             g_digest[ds_name] = {}
 
         feature_metrics = metrics[ds_name]
         for feature_name in feature_metrics:
             if feature_metrics[feature_name] is not None:
-                centroids: List = feature_metrics[feature_name].get(StC.STATS_CENTROIDS_KEY)
+                digest_dict: Dict = feature_metrics[feature_name].get(StC.STATS_DIGEST_COORD)
+                feature_digest = TDigest.from_dict(digest_dict)
                 if feature_name not in g_digest[ds_name]:
-                    g_digest[ds_name][feature_name] = TDigest()
-
-                for centroid in centroids:
-                    mean = centroid.get("m")
-                    count = centroid.get("c")
-                    g_digest[ds_name][feature_name].update(mean, count)
+                    g_digest[ds_name][feature_name] = feature_digest
+                else:
+                    g_digest[ds_name][feature_name] = g_digest[ds_name][feature_name].merge(feature_digest)
 
     return g_digest
 
 
-def compute_percentiles(g_digest: Dict[str, Dict[str, TDigest]], quantile_config: Dict, precision: int = 4) -> dict:
+def check_fastdigest_installed():
+    fastdigest, flag = optional_import("fastdigest")
+    if not flag:
```

Comment:
I think we should not do optional import in app_common.

Suggest either:
(1) move this file to app_opt
(2) create a new quantile_stats.py in app_opt and these logic should be moved there